### PR TITLE
[develop] Add a workflow for the data assimilation preprocessing tasks.

### PR DESCRIPTION
**Pull request by christinaholtNOAA**
## DESCRIPTION OF CHANGES: 

This PR is still a work in process as it requires stitching together the results of 4 other PRs:

- #636 Adding prepbufr task processing
- #638 Adding radar reflectivity processing
- #644 Adding lightning data processing
- [TBD] Adding da obs retrieval.

I am opening this one early to facilitate discussion about the direction of the workflow development for the RRFS project.

Here, I am adding untested YAML files that loosely follows what the end product for this PR will look like (hopefully). One YAML defines the default settings needed for these three data preprocessing tasks, then a second test YAML defines what it should look like to pull the data and run the tasks. There are still several details that need to be worked out concerning data retrieval.

Modifications are also planned in this PR to adopt the naming conventions set in Issue #633, but won't be made until the outstanding PRs are merged to develop, and then into this branch.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## TESTS CONDUCTED: 

- [ ] hera.intel
- [ ] orion.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] gaea.intel
- [ ] jet.intel
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [ ] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## CHECKLIST

- [ ] My code follows the style guidelines in the Contributor's Guide
- [ ] I have performed a self-review of my own code using the Code Reviewer's Guide
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

## LABELS (optional): 

A Code Manager needs to add the following labels to this PR: 
- [ ] Work In Progress
- [ ] bug
- [ ] enhancement
- [ ] documentation
- [ ] release
- [ ] high priority
- [ ] run_ci
- [ ] run_we2e_fundamental_tests
- [ ] run_we2e_comprehensive_tests
- [ ] Needs Cheyenne test 
- [ ] Needs Jet test 
- [ ] Needs Hera test 
- [ ] Needs Orion test 
- [ ] help wanted

### CODE REVIEW

File: parm/wflow/da_data_preproc.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+  task_process_radarref_#cycle_type#:
+    <<: *default_preproc
+    command: '&LOAD_MODULES_RUN_TASK_FP; "process_obs" "&JOBSdir;/JREGIONAL_PROCESS_RADARREF"'
+    ppn: 24
+    dependency:
+      or:
+        taskdep:
+          attrs:
+            task: get_da_obs_radar
```
Comment (christinaholtNOAA) [inline parm/wflow/da_data_preproc.yaml:39]: These names are placeholders for what comes of the retrieve data efforts underway by @mkavulich.

I am leaving the option here to either have this task run, or to have the script know where to look on disk for the sake of efficiency in the real-time runs.

File: tests/WE2E/test_configs/data_assimilation/config.process_obs.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+metadata:
+  description: |-
+    Tests the data retrieval and data preprocessing tasks that are
+    precursors to running any data assimialtion components. Datasets
+    include bufr, lightning, and radar reflectivity.
+
+workflow:
+  PREDEF_GRID_NAME: RRFS_CONUS_25km
+  DATE_FIRST_CYCL: '2019061500'
+  DATE_LAST_CYCL: '2019061500'
```
Comment (christinaholtNOAA) [inline tests/WE2E/test_configs/data_assimilation/config.process_obs.yaml:10]: I'll need some feedback from @EdwardSnyder-NOAA on what data has been staged here before this information can be nailed down.

Comment (EdwardSnyder-NOAA) [reply tests/WE2E/test_configs/data_assimilation/config.process_obs.yaml:10]: I placed the predefined grid data on Hera, Jet, and Orion for the following grids: `RRFS_NA_3km`, `RRFS_NA_13km`, and `RRFS_CONUS_3km`. The test lightning data is stored on Jet and Orion for the 2022-07-20 00Z case. We still need radar and buffer obs data for that date. Before I add those obs data, I just want to make sure that this is the case date we want to go with. My thinking was that this case could be the sample case we provide to the community for the release.

Comment (christinaholtNOAA) [reply tests/WE2E/test_configs/data_assimilation/config.process_obs.yaml:10]: @EdwardSnyder-NOAA I'm sorry for the delay on this one.

I think we should pose this question at a release meeting. More generally "is there an interesting case(s) that we should use for the release?"

**Comment by christinaholtNOAA** [comment]
The contents of this PR was incorporated into @mkavulich's branch for testing observation data retrieval.

**Pull request closed without merging**